src/Strategy/Dart/PubDeps.hs

```diff
@@ -0,0 +1,215 @@
+    }
+
+data PubDepPackage = PubDepPackage
+  { pubDepPackageName :: PackageName
+  , pubDepPackageVersion :: Maybe VerConstraint
+  , pubDepPackageDeps :: Maybe (Set PackageName)
+  , pubDepPackageIsDirect :: Bool
+  }
+  deriving (Generic, Show, Eq, Ord)
+
+-- | Parse package name.
+-- Ref: https://dart.dev/tools/pub/pubspec#name
+parsePackageName :: Parser PackageName
+parsePackageName = PackageName . toText <$> many (alphaNumChar <|> char '_')
+
+-- | Parse package version.
+-- Ref: https://dart.dev/tools/pub/pubspec#version
+parsePackageVersion :: Parser VerConstraint
+parsePackageVersion = CEq . toText <$> many (alphaNumChar <|> char '.' <|> char '-' <|> char '+')
+
+-- | Parses Pub Package Entry.
+--  - pkg_name 1.0.0 [pkg_dep_one]
+--  - pkg_name 1.0.0
+--  - pkg_name 1.0.0 [pkg_dep_one, pkg_dep_two]
+parsePubDepPackage :: Bool -> Parser PubDepPackage
+parsePubDepPackage isDirectDep = do
+  packageName <- lexeme $ symbol "-" *> lexeme parsePackageName
+  packageVersion <- lexeme parsePackageVersion
+
+  -- package may not have any dependencies
+  packageSubDeps <- optional (between (symbol "[") (symbol "]") (Set.fromList <$> sepBy parsePackageName (symbol " ")))
+
+  _ <- optional newline
+  pure $ PubDepPackage packageName (Just packageVersion) packageSubDeps isDirectDep
+
+depsCmdOutputParser :: Parser [PubDepPackage]
+depsCmdOutputParser = parseDeps <* eof
+  where
+    parseDeps = do
+      directDeps <- skipManyTill anySingle (symbol "dependencies:") *> many (parsePubDepPackage True)
+      devDeps <- optional $ chunk "dev dependencies:" *> newline *> many (parsePubDepPackage True)
+
+      -- Since, this strategy requires lockfile - dependency override information is redundant,
+      -- as lock file already produces resolved dependency source.
+      _ <- optional $ chunk "dependency overrides:" *> newline *> many (parsePubDepPackage False)
+      transitiveDeps <- optional $ chunk "transitive dependencies:" *> newline *> many (parsePubDepPackage False)
+
+      pure $
+        directDeps
+          ++ fromMaybe [] devDeps
+          ++ fromMaybe [] transitiveDeps
+
+isPackageSupported :: PubLockContent -> PackageName -> Bool
+isPackageSupported lockContent pkg = maybe False isSupported $ Map.lookup pkg (packages lockContent)
+
+buildGraph :: PubLockContent -> [PubDepPackage] -> Graphing Dependency
+buildGraph lockContent pkgs = Graphing.promoteToDirect isDirectDependency graphPackagesWithEdges
+  where
+    isDirectDependency :: Dependency -> Bool
+    isDirectDependency dep = dependencyName dep `elem` (dependencyName . pkgToDependency <$> reportableDirectPackages)
+
+    graphPackagesWithEdges :: Graphing Dependency
+    graphPackagesWithEdges = foldr (uncurry edge) graphPackagesWithoutEdges edges
+
+    edges :: [(Dependency, Dependency)]
+    edges = concatMap edgesOf pkgs
+
+    edgesOf :: PubDepPackage -> [(Dependency, Dependency)]
+    edgesOf pkg =
+      (pkgToDependency $ pubDepPackageName pkg,) . pkgToDependency
```

Comment:
I'd prefer explicit parens here; my mental parser breaks on the combination of the `,` operator sectioning and `$`
```suggestion
      (pkgToDependency (pubDepPackageName pkg) ,) . pkgToDependency
```